xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/renderer/xhtml/link/DocumentXHTMLLinkTypeRenderer.java

```diff
@@ -90,6 +102,35 @@ protected String computeLabel(ResourceReference reference)
         return this.linkLabelGenerator.generate(reference);
     }
 
+    private URITitleGenerator getTitleGenerator(ResourceReference reference)
+    {
+        URITitleGenerator titleGenerator = this.defaultTitleGenerator;
+        if (this.componentManager.hasComponent(URITitleGenerator.class, reference.getType().getScheme())) {
+            try {
+                titleGenerator = this.componentManager.getInstance(URITitleGenerator.class,
+                    reference.getType().getScheme());
+            } catch (Exception e) {
+                logger.warn("Error while loading component for generating URI title: [{}]",
+                    ExceptionUtils.getRootCauseMessage(e));
+                logger.debug("Full stack trace: ", e);
```

Comment:
I would write:

```
String message = "Error while loading component for generating URI title";
if (logger.isDebugEnabled()) {
    logger.debug(message, e);
} else {
    logger.warn(String.format("%s: [{}]", message), ExceptionUtils.getRootCauseMessage(e));
}
```

Reply:
In addition you should also log the reference and explain which component we're talking about.